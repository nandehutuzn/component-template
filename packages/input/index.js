import ZnInput from './src/input.vue'

ZnInput.install = Vue => {
  Vue.component(ZnInput.name, ZnInput)
}

export default ZnInput