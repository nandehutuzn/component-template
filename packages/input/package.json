{
  "name": "zn-input",
  "version": "0.0.1",
  "description": "zn-input component",
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "author": "男的糊涂 <120902259@qq.com>",
  "license": "MIT",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "del": "rimraf dist"
  },
  "keywords": []
}
