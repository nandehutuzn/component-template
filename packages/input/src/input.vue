<template>
  <div>
    <input v-bind="$attrs" :type="type" :value="value" @input="handleInput">
  </div>
</template>

<script>
export default {
  name: 'ZnInput',
  inheritAttrs: false,
  props: {
    value: String,
    type: {
      type: String,
      default: 'text'
    }
  },
  methods: {
    handleInput(evt) {
      this.$emit('input', evt.target.value)
      const findParent = parent => {
        while(parent) {
          if(parent.$options.name === 'ZnFormItem') {
            break
          } else {
            parent = parent.$parent
          }
        }

        return parent
      }

      const parent = findParent(this.$parent)
      if(parent) {
        parent.$emit('validate')
      }
    }
  }
}
</script>

<style scoped>
</style>