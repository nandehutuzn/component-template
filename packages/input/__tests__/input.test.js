import { mount } from '@vue/test-utils'
import input from '../src/input.vue'

describe('ZnInput', () => {
  test('input-text', () => {
    const wrapper = mount(input)

    expect(wrapper.html()).toContain('input type="text"')
  })

  test('input-password', () => {
    const wrapper = mount(input, {
      propsData: {
        type: 'password'
      }
    })

    expect(wrapper.html()).toContain('input type="password"')
  })

  test('input-password2', () => {
    const wrapper = mount(input, {
      propsData: {
        type: 'password',
        value: 'admin'
      }
    })

    expect(wrapper.props('value')).toBe('admin')
  })

  test('input-snapshot', () => {
    const wrapper = mount(input, {
      propsData: {
        type: 'text',
        value: 'admin'
      }
    })

    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})