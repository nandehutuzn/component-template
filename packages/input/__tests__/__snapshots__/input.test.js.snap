// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZnInput input-snapshot 1`] = `
<div>
  <input
    type="text"
  />
</div>
`;
